src/uct/ib/dc/dc_mlx5.c

```diff
@@ -879,17 +886,14 @@ uct_dc_mlx5_iface_dcis_create(uct_dc_mlx5_iface_t *iface,
     uct_dc_dci_t *dci;
 
     ucs_array_init_dynamic(&iface->tx.dcis);
-    /* Reserving max size as a temporary solution due to pointer-copy issue in ucs_array
-       TODO: fix when ucs_array_t supports complex types 
-    */
-    status = uct_dc_mlx5_iface_resize_and_fill_dcis(
-            iface, UCT_DC_MLX5_IFACE_MAX_DCI_POOLS * iface->tx.ndci);
+    status = uct_dc_mlx5_iface_resize_and_fill_dcis(iface, 1);
```

Comment:
why do we need one DCI? can we not call this function at all?

Reply:
`uct_dc_mlx5_iface_create_dci` is relying on the fact that there is enough space in `dcis` array